app/scripts/context.ts

```diff
@@ -0,0 +1,9 @@
+export enum METAMASK {
+  CONTENTSCRIPT = 'metamask-contentscript',
+  UI = 'metamask-ui',
+  INPAGE = 'metamask-inpage',
+  BACKGROUND = 'metamask-background',
+  EXTERNAL = 'metamask-external',
+  PHISHING_WARNING_PAGE = 'metamask-phishing-warning-page',
+  PROVIDER = 'metamask-provider',
+}
```

Comment:
I changed METAMASK to METAMASK_CTX to be clearer to your request 039129b65 (without "logging" because this is used outside of only logging).

As for enums, originally (an hour ago 01dd34ba75d6870b9ce721141e430c7a6f1d4de3) it was a regular `const` but @mcmire [request](https://github.com/MetaMask/metamask-extension/pull/20660#discussion_r1310670036) enums.

I don't think neither is significantly better than the other, so it depends on whether @mcmire has a strong objection to `const` vs enums or not?

Reply:
I've been recommending that people use enums because they tend to be easier to understand and parse for people who are newer to TypeScript and so they tend to be the thing they gravitate to. It's true that creating an object + a type tends to be a more flexible approach than using `enum`, but in my experience we rarely need that flexibility. Also, in response to @legobeat, the `enum` keyword already creates both an object and a type for you, so it should already work in a non-TS perspective. In a TS context, if you just need the type you can use `import type` and if you need both you can just use `import`, so there shouldn't be any differences from a usability perspective.

That said, if we choose to use object + a type, I would recommend naming the object using a singular name rather than a plural name. The reason is that from a type perspective, an enum is essentially a type union, and type unions tend to be named with a singular (e.g. `Color` vs `Colors`).